stdlib/stack_trace.mli

```diff
@@ -0,0 +1,141 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Stack traces.
+
+    Facilities for getting the current stack trace, printing and inspecting
+    them.
+
+    @since 4.08 *)
+
+type t
+(** The type of stack traces.
+
+    Stack traces cannot be marshalled. If you need marshalling, you should use
+    the array returned by the {!slots} function. *)
+
+val current: int -> t
+(** [current n] returns a description of the stack trace on the current program
+    point (for the current thread), with at most [n] entries. *)
+
+val to_string: t -> string
+(** Return a string listing the program locations contained in the stack
+    trace. *)
+
+val length: t -> int
+(** [length st] returns the number of (non-inlined) slots in the stack trace
+    [st]. *)
+
+module Slot : sig
```

Comment:
I don't find `Slot` to be a great name. It seems `Java` simply call these [StackTraceElement](https://docs.oracle.com/javase/10/docs/api/java/lang/StackTraceElement.html) and `.NET` [StackFrame](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stackframe?view=netframework-4.7.2). Maybe `Frame` or `Site` ? I would tend to favour the latter.

Reply:
I have a slight preference for `Frame` rather than `Site`, which sounds a bit too generic to me.

Reply:
I would suggest that these are called "frames", which matches standard terminology in e.g. `gdb`, but the function that obtains a location from a frame is called "`call_site`" or similar.

Reply:
I trust @mshinwell to have the right name.

Reply:
OK, I changed the name of the module to `Frame` and the function `Stack_trace.slots` to `Stack_trace.frames`.